File: Cmdlets/RegisterCultureAndRegionInfoCommand.vb
Imports System.Globalization
Imports System.Management.Automation

<Cmdlet("Register", "CultureAndRegionInfo")>
Public Class RegisterCultureAndRegionInfoCommand
	Inherits CultureAndRegionInfoCommandBase

	<Parameter(Mandatory:=True)>
	Public Property Name() As String

	Protected Overrides Sub BeginProcessing()
		Dim builder As CultureAndRegionInfoBuilder = CultureAndRegionInfoDictionary(Name)

		builder.Register()
	End Sub
End Class

File: C#/Cmdlets/Cmdlets/RegisterCultureAndRegionInfoCommand.cs
using System.Globalization;
using System.Management.Automation;

namespace TIKSN.Cmdlets
{
    [Cmdlet("Register", "CultureAndRegionInfo")]
    public class RegisterCultureAndRegionInfoCommand : CultureAndRegionInfoCommandBase
    {
        [Parameter(Mandatory = true)]
        public string Name { get; set; }

        protected override void BeginProcessing()
        {
            CultureAndRegionInfoBuilder builder = CultureAndRegionInfoDictionary[Name];

            builder.Register();
        }
    }
}